Add defaults chart upgrade button to dashboard

jordojordo opened the pull request:
Fix #655 
Fix #700 

This adds a upgrade button for the `kubewarden-defaults` chart on the Dashboard view. Refactored a few things to reuse the functions used to determine if an upgrade is available for the controller for a more agnostic approach. Also fixed the alignment of the badges.

__With upgradable controller and defaults chart__

![kw-dashboard-upgrades](https://github.com/rancher/kubewarden-ui/assets/40806497/9763a407-2ead-480f-81b7-b944df56ba56)

__With only upgradable defaults__

![kw-defaults-upgrades](https://github.com/rancher/kubewarden-ui/assets/40806497/dc971c40-b09e-44ed-a150-43c9ac43ebcb)

kravciak commented:
Hi, I noticed few issues:
 - mouse cursor does not update when I hover over the upgrade buttons - it (usually) keeps shape from last element (arrow, hand, text select, ...) - is this rancher issue?
 - should we disable defaults upgrade until controller is upgraded (within the same or lower app version)
 - button names - I suggest small change to be consistent:
```
App Version: v1.11.0
App Upgrade: v1.12.0 - Controller: 2.0.11
App Upgrade: v1.12.0 - Defaults: 2.0.0
```

kravciak commented:
I found better explanation in [our docs](https://docs.kubewarden.io/reference/upgrade-path):
```
1. Stack version compatibility among components (https://docs.kubewarden.io/reference/upgrade-path#stack-version-compatibility-among-components)
kubewarden-crds chart, kubewarden-controller chart, kubewarden-defaults chart, any manually deployed policy-server image, and kwctl should run the same MAJOR/MINOR. The PATCH version can increase independently though.
Therefore, if the kubewarden-controller version running is 1.1.x, the policy-servers and kwctl version in use should be 1.1.x as well.

2. Upgrade order (https://docs.kubewarden.io/reference/upgrade-path#upgrade-order)
Kuberwarden users should upgrade the stack starting with the kubewarden-crds chart, then the kubewarden-controller chart. After that, upgrade the policy-server (via kubewarden-defaults chart or bumping the images of custom ones) and kwctl.
```

Buttons don't follow those rules:
 - it offers all controller upgrades available, ignoring compatibility requirement with defaults
 - it does not care about upgrade order

jordojordo commented:
@kravciak  I've updated the defaults chart upgrade button to only be shown when it is compatible with the installed controller chart. But this leads me to a thought, should we limit the ability to update the controller chart when the defaults chart hasn't been upgraded to match the installed version of the controller?

As for the mouse cursor not updating when hovering over the buttons, I do not see this behavior, could you take a video?

Here's a set of app upgrades:

![appv-1 10 0-defaultsupgrade](https://github.com/rancher/kubewarden-ui/assets/40806497/219655bf-0488-488a-94ee-f36bcf2a40f5)

jordojordo commented:
I removed the upgrade button for the controller chart when the versions are out of sync. There will be a banner displayed informing the user that the defaults chart must be updated to match the appVersion of the controller chart.

___appVersion mismatch___
![defaults-needsupgrade](https://github.com/rancher/kubewarden-ui/assets/40806497/2af28b62-0f15-451c-9a46-11026c124c73)

___appVersion satisfied___
![defaults-afterupgrade](https://github.com/rancher/kubewarden-ui/assets/40806497/b4fccf2b-e4ba-4f69-a83c-2de7cf1ea0ca)

kravciak reviewed (changes requested):
I installed kubewarden `controller v2.8` & `defaults 1.9.3`. Everything works fine.
Then I upgraded controller (from rancher apps menu) into `2.0.9`.

After upgrade kubewarden UI fails to load and I see this error in console:


```
vue.min.js:11 TypeError: Invalid version. Must be a string. Got type "undefined".
    at new u (semver.js:19:13)
    at Object.i [as major] (major.js:2:29)
    at a.getValidUpgrade (DashboardView.vue:291:1)
    at a.checkUpgradeAvailable (DashboardView.vue:376:1)
    at a.defaultsUpgradeAvailable (DashboardView.vue:223:1)
    at e.get (vue.min.js:11:27998)
    at e.evaluate (vue.min.js:11:28994)
    at a.defaultsUpgradeAvailable (vue.min.js:11:31170)
    at a.m (DashboardView.vue:1:2390)
    at e._render (vue.min.js:11:45499)
```

Then
 - I upgraded to controller 2.0.10 from rancher apps -> broken
 - downgraded to controller 2.0.8 from rancher apps -> UI works again
 - upgraded to 2.0.10 from kubewarden overview page -> broken

jordojordo commented:
The issue was caused by an undefined version in the `getValidUpgrade` method. I added an extra condition in the beginning to exit the method when either `currentVersion` or `upgradeVersion` are undefined.
Also I fixed the pre-release check in the `appVersionSatisfies` method to use the pre-release setting.

jordojordo commented:
If the users are out of sync with an old defaults, the upgrade button should install defaults step by step to be in accordance with our upgrade strategy.
